core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -141,6 +136,18 @@ private[spark] abstract class Task[T](
   def executorDeserializeTime: Long = _executorDeserializeTime
 
   /**
+   * Collect the latest values of accumulators used in this task. If the task failed,
+   * filter out the accumulators whose values should not be included on failures.
+   */
+  def collectAccumulatorUpdates(taskFailed: Boolean = false): Seq[AccumulableInfo] = {
+    if (context != null) {
```

Comment:
Just curious: under which circumstances would this method be called when `context == null`? Would that only happen if `run()` failed immediately before TaskContext initialization?